docs/contributing/docs-and-website-components.md

```diff
@@ -0,0 +1,409 @@
+
+Information about authoring in Markdown and styling components on the site is also listed.
+
+## Globally available components
+
+A variety of components have been written to help with formatting code and content across the blog and the docs and don't need to be imported.
+
+---
+
+### Guide List
+
+The `<GuideList />` is a component that renders a list of child docs nested below the current doc in the site's information architecture. It is often used on overview pages like the [Headless CMS](/docs/headless-cms/) guide where many other pages are nested below it, to show what a docs section contains.
+
+#### Usage
+
+The Guide List component takes one prop:
+
+- `slug` - the value of which is already available on every page's context on Gatsbyjs.org by default
+
+The component can be used like this:
+
+```markdown:title=docs/headless-cms
+---
+title: Headless CMS
+---
+
+Overview information about headless CMSs...
+
+<GuideList slug={props.slug} /> // highlight-line
+```
+
+#### Sample
+
+When rendered in a page, the Guide List looks like this:
+
+<GuideList slug="/docs/headless-cms/" />
+
+---
+
+### Egghead Embed
+
+To embed video content from [Egghead](https://egghead.io) on the site, there is an `<EggheadEmbed />` component that adds an iframe with the video inline with other content.
+
+#### Usage
+
+The Egghead Embed component takes two props:
+
+- `lessonLink` - the URL of the lesson from Egghead
+- `lessonTitle` - the name of the lesson that is used to add more information to the embedded iframe.
+
+It can be used like this:
+
+```markdown:title=docs/using-gatsby-image
+---
+title: Using Gatsby Image
+---
+
+This is how you use `gatsby-image`...
+
+<!-- highlight-start -->
+
+<EggheadEmbed
+  lessonLink="https://egghead.io/lessons/gatsby-use-gatsby-image-s-graphql-fragments-for-blurred-up-and-traced-svg-images"
+  lessonTitle="Use gatsby-image's GraphQL fragments for blurred-up and traced SVG images"
+/>
+
+<!-- highlight-end -->
+```
+
+#### Sample
+
+Rendered, it looks like this:
+
+<EggheadEmbed
+  lessonLink="https://egghead.io/lessons/gatsby-use-gatsby-image-s-graphql-fragments-for-blurred-up-and-traced-svg-images"
+  lessonTitle="Use gatsby-image's GraphQL fragments for blurred-up and traced SVG images"
+/>
+
+---
+
+### Pull Quote
+
+The `<Pullquote />` component is used to call out a quote in the blog. It applies borders and styles that make a section of the content pop out to readers.
+
+#### Usage
+
+The Pull Quote component takes one prop, and uses the children it wraps to populate its inner content:
+
+- `citation` - the reference of the person or entity that made the quoted statement
+
+It is used like this:
+
+```markdown:title=blog/new-announcement.md
+---
+title: Big News!
+author: Jay Gatsby
+---
+
+<!-- highlight-start -->
+<Pullquote citation="Winston Churchill">
```

Comment:
The border around this pullquote is overlapping the TOC (which probably answers my question above):

<img width="845" alt="pullquote example from Winston Churchill" src="https://user-images.githubusercontent.com/1045233/67515227-4077ba80-f66c-11e9-8330-8c9508e94739.png">

Reply:
Yeah, do you think we should not include it or style it to shrink it down in this example?

Reply:
I think including it is a good idea, and solving the issue with the TOC could fix it for usage later on. Does wrapping it in an element help, at all? Or we could introduce some CSS to make it render better in docs?

Reply:
Wrapping it requires new styles on the parent container, since it's a component I figured adding a prop would make sense to override the extending width and force it to match the parent container width, I named the prop `narrow` which is either true or false.

![image](https://user-images.githubusercontent.com/21114044/67780562-e2602400-fa2b-11e9-917b-3bda09d0d53f.png)

Reply:
Sounds great!

Reply:
i would avoid additional options, because if someone use it in docs then he has always remember to set the option, and this is need to be documented etc.

more solutions:

a) how about that this is always fit to the container width? 

b) or only between > 1000px and <1200px
- the setting is to be outside the container width when it is only above >1000px - but the TOC becomes visible at >1200px

c) how about: if this only is more wide when it is an blog post, you can distinct it from a doc if it is an child of `<section class='post-body'>`